### PR TITLE
Translation helper improvement (added html parameter)

### LINKED ISSUE #7798: Translation string replacement errors
A huge and wonderful effort to set up translation strings for site-wide texts has been going on for some time in #7416 and other issues -- many thanks to those who've helped!

I noticed that when a non-english language is selected, we now have some issues with the translation icons appearing:

image

Some are parsing errors causing for example the text `" data-placement="bottom" href="/dashboard">Dashboard` to appear. Others simply add the globe icon causing items not to fit in the layout anymore. For comparison, here is the dashboard in English:

image

You can see this happening by clicking on this link which will change the language of the `stable` server so you can see the translations turned on (this for spanish):

https://stable.publiclab.org/change_locale/es

I'd like to ask for help in solving these parsing and layout issues! They should be pretty simple and you should be able to test them locally by setting a language locally, for example with:

http://localhost:3000/change_locale/es

Thanks, everyone!

### PR DESCRIPTION
**Pull request by imajit**
Part of #9686 
Fixes #7798 #9494 
As the translation function was called inside the HTML tag, there were rendering issues; I fixed it by adding an `HTML = true` optional parameter in the function, now we can pass `html=false` in cases where the function is called inside the HTML tag.
Current UI
Screenshot from 2021-05-30 17-22-52

After changes 
Screenshot from 2021-05-30 17-23-32

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -154,12 +154,12 @@ def current_page?(page)
     'is-active' if request.path == page
   end
 
-  def translation(key, options = {})
+  def translation(key, options = {},html = true)
```
[the remark this answers is not in the document]
Comment (imajit) [reply app/helpers/application_helper.rb]: Silly me

[the remark this answers is not in the document]
Comment (jywarren) [reply app/helpers/application_helper.rb]: Don't take codeclimate bot /too/ seriously, it needs to learn to be friendlier! But thank you!

**Comment by jywarren** [comment]
Ah it looks like perhaps the conflicts need to be resolved to trigger the tests?

**Comment by jywarren** [comment]
Thanks, this is looking great so far!

**Comment by RuthNjeri** [comment]
This looks great @imajit 🎉 ...and thanks for replacing the `t(` with `translation(`

**Comment by RuthNjeri** [comment]
The issue you have referenced #9696 has been closed, is there another main planning issue you can reference?

**Review by RuthNjeri — APPROVED** [review]
LGTM

**Comment by imajit** [comment]
> The issue you have referenced #9696 has been closed, is there another main planning issue you can reference?

No, I guess it got closed when I tagged it in a different PR. 
I reopened it

**Review by Tlazypanda — APPROVED** [review]
@imajit LGTM 🎉

**Review by cesswairimu — APPROVED** [review]
Looking Great :tada: thanks @imajit

**Comment by cesswairimu** [comment]
merged..lets confirm on stable after build.. thanks everyone

**Pull request merged**